projects/storefrontlib/cms-components/product/carousel/product-carousel-item/product-carousel-item.component.spec.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  ChangeDetectionStrategy,
+  Directive,
+  Injector,
+  Input,
+  Pipe,
+  PipeTransform,
+  SimpleChange,
+} from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import {
+  I18nTestingModule,
+  ProductService,
+  RoutingService,
+} from '@spartacus/core';
+import {
+  CmsComponentData,
+  OutletDirective,
+  OutletModule,
+  PageComponentModule,
+  ProductListItemContext,
+  ProductListItemContextSource,
+} from '@spartacus/storefront';
+import { ProductCarouselItemComponent } from './product-carousel-item.component';
+import {
+  AddToCartComponent,
+  AddToCartModule,
+} from '@spartacus/cart/base/components/add-to-cart';
+import { StoreModule } from '@ngrx/store';
+import { BehaviorSubject } from 'rxjs';
+import { By } from '@angular/platform-browser';
+
+@Pipe({
+  name: 'cxUrl',
+})
+class MockUrlPipe implements PipeTransform {
+  transform() {}
+}
+
+class MockRoutingService {}
+class MockProductService {}
+
+@Directive({
+  selector: '[cxOutlet]',
+})
+class MockOutletDirective implements Partial<OutletDirective> {
+  @Input() cxOutlet: string;
+}
+
+describe('ProductCarouselItemComponent in product-carousel', () => {
+  let component: ProductCarouselItemComponent;
+  let componentInjector: Injector;
+  let fixture: ComponentFixture<ProductCarouselItemComponent>;
+
+  const mockProduct = {
+    name: 'Test product',
+    nameHtml: 'Test product',
+    summary: 'Test summary',
+    code: '1',
+    averageRating: 4.5,
+    stock: {
+      stockLevelStatus: 'inStock',
+    },
+    price: {
+      formattedValue: '$100,00',
+    },
+    images: {
+      PRIMARY: {},
+    },
+  };
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [RouterTestingModule, I18nTestingModule, OutletModule],
+        declarations: [
+          ProductCarouselItemComponent,
+          MockUrlPipe,
+          MockOutletDirective,
+        ],
+        providers: [
+          {
+            provide: RoutingService,
+            useClass: MockRoutingService,
+          },
+          {
+            provide: ProductService,
+            useClass: MockProductService,
+          },
+        ],
+      })
+        .overrideComponent(ProductCarouselItemComponent, {
+          set: { changeDetection: ChangeDetectionStrategy.Default },
```

Comment:
Was it required to override the change detection strategy for the unit tests to work?

Reply:
No, I think I can omit this.